tests/hardhat/Fork/comptrollerXVSAddressesUpgrade.ts

```diff
@@ -0,0 +1,110 @@
+const OLD_SETTER_FACET = "0xF2b7D75557B75a878E997934014E95Dd089B5f24";
+const OLD_REWARD_FACET = "0x71e7AAcb01C5764A56DB92aa31aA473e839d964F";
+const UNITROLLER = "0xfD36E2c2a6789Db23113685031d7F16329158384";
+const zeroAddr = "0x0000000000000000000000000000000000000000";
+
+forking(34340887, () => {
+  let diamond: Diamond;
+  let owner: SignerWithAddress;
+
+  if (process.env.FORK === "true" && process.env.FORKED_NETWORK === "bscmainnet") {
+    before(async () => {
+      await impersonateAccount(Owner);
+      owner = await ethers.getSigner(Owner);
+
+      diamond = await ethers.getContractAt("Diamond", UNITROLLER);
+    });
+
+    it("pre checks", async () => {
+      const rewardFacet = await ethers.getContractAt("RewardFacet", diamond.address);
+
+      expect(await rewardFacet.getXVSVTokenAddress()).to.be.equal("0x151B1e2635A717bcDc836ECd6FbB62B674FE3E1D");
+      expect(await rewardFacet.getXVSAddress()).to.be.equal("0xcF6BB5389c92Bdda8a3747Ddb454cB7a64626C63");
+    });
+
+    it("upgrade checks", async () => {
+      const oldSetterFacet = await ethers.getContractAt("SetterFacet", diamond.address);
+      await expect(
+        oldSetterFacet.connect(owner)._setXVSToken("0x151B1e2635A717bcDc836ECd6FbB62B674FE3E1D"),
+      ).to.be.revertedWith("Diamond: Function does not exist");
+      await expect(
+        oldSetterFacet.connect(owner)._setXVSVToken("0x151B1e2635A717bcDc836ECd6FbB62B674FE3E1D"),
+      ).to.be.revertedWith("Diamond: Function does not exist");
+
+      const setterFacetFactory = await ethers.getContractFactory("SetterFacet");
+      const newSetterFacet: SetterFacet = await setterFacetFactory.deploy();
+      await newSetterFacet.deployed();
+
+      const addXVSAddressFunctionSignature = newSetterFacet.interface.getSighash(
+        newSetterFacet.interface.functions["_setXVSToken(address)"],
+      );
+      const addXVSVTokenAddressFunctionSignature = newSetterFacet.interface.getSighash(
+        newSetterFacet.interface.functions["_setXVSVToken(address)"],
+      );
+
+      const rewardFacetFactory = await ethers.getContractFactory("RewardFacet");
+      const newRewardFacet: RewardFacet = await rewardFacetFactory.deploy();
+      await newRewardFacet.deployed();
+
+      const existingSetterFacetFunctions = await diamond.facetFunctionSelectors(OLD_SETTER_FACET);
+      const existingRewardFacetFunctions = await diamond.facetFunctionSelectors(OLD_REWARD_FACET);
+
+      const cut = [
+        {
+          facetAddress: newSetterFacet.address,
+          action: 0,
+          functionSelectors: [addXVSAddressFunctionSignature, addXVSVTokenAddressFunctionSignature],
+        },
+        {
+          facetAddress: newSetterFacet.address,
+          action: 1,
+          functionSelectors: existingSetterFacetFunctions,
+        },
+        {
+          facetAddress: newRewardFacet.address,
+          action: 1,
+          functionSelectors: existingRewardFacetFunctions,
+        },
+      ];
```

Comment:
Maybe move the numbers to an enum? e.g.

```ts
enum FacetCutAction {
     Add,
     Replace,
     Remove
}
```

```ts
const cut = [
  {
    facetAddress: newSetterFacet.address,
    action: FacetCutAction.Add,
    functionSelectors: [addXVSAddressFunctionSignature, addXVSVTokenAddressFunctionSignature],
  },
  {
    facetAddress: newSetterFacet.address,
    action: FacetCutAction.Replace,
    functionSelectors: existingSetterFacetFunctions,
  },
  {
    facetAddress: newRewardFacet.address,
    action: FacetCutAction.Replace,
    functionSelectors: existingRewardFacetFunctions,
  },
];
```

I'd even store this enum in some utility file so that we could reuse it in tests